### PR TITLE
Add permission dialog

### PR DESCRIPTION
**Pull request by HugoGresse**
This is related to https://github.com/Instant-Visio/InstantVisio-WebApp/issues/161

It add a new dialog to check camera and microphone access. 

This is a first version, in another PR I'll split the VideoCall to have the dialog shown before doing anything related to the real video call (loading daily.co for eg) and also adding other check on the browser

ezgif-6-2e91c75a25c6

### CODE REVIEW

File: src/hooks/useCameraMicrophonePermission.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                video: true,
+                audio: true,
+            },
+            function () {
+                setResult(STATE_GRANTED)
+            },
+            function (err) {
+                console.log('err', err)
+                setResult(STATE_DENIED)
+                // if(err === PERMISSION_DENIED) {
```
Comment (ntchou) [inline src/hooks/useCameraMicrophonePermission.js:21]: we can remove these useless comments

File: src/pages/VideoCall/permissions/CameraMicrophonePermission.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    const permissionResult = useCameraMicrophonePermission('camera')
+
+    useEffect(() => {
+        if (permissionResult === STATE_GRANTED) {
+            onGranted()
+        }
+    }, [permissionResult, onGranted])
+
+    return (
+        <>
```
Comment (ntchou) [inline src/pages/VideoCall/permissions/CameraMicrophonePermission.js:19]: Fragment tag is not necessary

File: src/pages/Home/Description.js [hunk 3]
```diff
@@ -142,7 +142,8 @@ function Description() {
             <Baseline>
                 <Trans i18nKey="common:homeBaseline">
                     {/* prettier-ignore */}
-                    Joignez un proche en visio, en un clic, <span>gratuitement.</span>
+                    Joignez un proche en visio, en un clic,{' '}
+                    <span>gratuitement.</span>
```
Comment (ntchou) [inline src/pages/Home/Description.js:146]: `{' '}` breaks the translation
If you check the  homepage `gratuitement` has disappeared

File: src/pages/VideoCall/permissions/CapabilitiesDialog.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        if (Object.values(grantedMap).filter((value) => !!value)) {
+            setCheckPass(true)
+            setTimeout(() => {
+                setDialogDisplay(false)
+            }, 5000)
+        }
+    }
+
+    useEffect(() => {
+        window.iv = window.iv || {}
```
Comment (ntchou) [inline src/pages/VideoCall/permissions/CapabilitiesDialog.js:34]: why don't you define `window.iv = {}` directly?

File: src/pages/VideoCall/permissions/Check.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        flex-basis: auto;
+        margin-right: 12px;
+        margin-left: 4px;
+        width: 30px;
+        height: 30px;
+    }
+`
+
+const iconSize = 1.5
+const Check = ({ state, i18nKey, i18nInstance }) => {
```
Comment (ntchou) [inline src/pages/VideoCall/permissions/Check.js:31]: you can import i18n from '../../../i18n/i18n' and use it like this. That avoid you to inject i18nInstance
```i18n.t(`videocall:${i18nKey}.waiting`)```

File: src/hooks/useCameraMicrophonePermission.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import React, { useEffect } from 'react'
+
+export const STATE_WAITING = 0
+export const STATE_GRANTED = 1
+export const STATE_DENIED = 2
+
+export default function useCameraMicrophonePermission(name) {
+    const [result, setResult] = React.useState(STATE_WAITING)
+    useEffect(() => {
+        navigator.getUserMedia(
```
Comment (ntchou) [inline src/hooks/useCameraMicrophonePermission.js:10]: caution `navigator.getUserMedia` is deprecated use `navigator.mediaDevices.getUserMedia()` instead.
Error on firefox 75
image

File: src/pages/VideoCall/permissions/CapabilitiesDialog.js (hunk 4 above)
Comment (HugoGresse) [reply src/pages/VideoCall/permissions/CapabilitiesDialog.js:34]: it's a good practice when touching global object linked to the window.

File: src/pages/VideoCall/permissions/Check.js (hunk 5 above)
Comment (HugoGresse) [reply src/pages/VideoCall/permissions/Check.js:31]: you're right, Check was abtract to any logic but I've changed my mind in the meantime.

File: src/pages/VideoCall/permissions/CapabilitiesDialog.js (hunk 4 above)
Comment (ntchou) [reply src/pages/VideoCall/permissions/CapabilitiesDialog.js:34]: You're right. I used to check if object exists. i take note of the tip

File: public/locales/fr/videocall.json [hunk 7]
```diff
@@ -12,7 +12,7 @@
         "dialog": {
             "title": "Permissions",
             "description": "Pour assurer que l'appel vidéo marche, nous avons besoin des permissions suivantes:",
-            "autoclose": "Tout semble fonctionner, cette fenêtre via disparaitre dans quelques seconds."
+            "autoclose": "Tout semble fonctionner, cette fenêtre via disparaitre dans quelques secondes."
```
Comment (Abibibi) [inline public/locales/fr/videocall.json:15]: Please replace "via disparaitre" with "va disparaître"

**Comment by HugoGresse** [comment]
It's should be good to merge, let me know if you can check a last time.

**Pull request merged**